lib/ansible/modules/storage/netapp/na_ontap_net_port.py

```diff
@@ -0,0 +1,233 @@
+        net_port_attributes = netapp_utils.zapi.NaElement('net-port-info')
+        net_port_attributes.add_new_child('node', self.node)
+        net_port_attributes.add_new_child('port', self.port)
+        query = netapp_utils.zapi.NaElement('query')
+        query.add_child_elem(net_port_attributes)
+        net_port_info.add_child_elem(query)
+        result = self.server.invoke_successfully(net_port_info, True)
+        return_value = None
+
+        if result.get_child_by_name('num-records') and \
+                int(result.get_child_content('num-records')) >= 1:
+
+            net_port_attributes = result.get_child_by_name('attributes-list').\
+                get_child_by_name('net-port-info')
+            return_value = {
+                'node': net_port_attributes.get_child_content('node'),
+                'port': net_port_attributes.get_child_content('port'),
+                'mtu': net_port_attributes.get_child_content('mtu'),
+                'autonegotiate_admin': net_port_attributes.get_child_content(
+                    'is-administrative-auto-negotiate'),
+                'duplex_admin': net_port_attributes.get_child_content(
+                    'administrative-duplex'),
+                'speed_admin': net_port_attributes.get_child_content(
+                    'administrative-speed'),
+                'flowcontrol_admin': net_port_attributes.get_child_content(
+                    'administrative-flowcontrol'),
+            }
+        return return_value
+
+    def modify_net_port(self):
+        """
+        Modify a port
+        """
+        port_obj = netapp_utils.zapi.NaElement('net-port-modify')
+        port_obj.add_new_child("node", self.node)
+        port_obj.add_new_child("port", self.port)
+        # The following options are optional.
+        # We will only call them if they are not set to None
+        if self.mtu:
+            port_obj.add_new_child("mtu", self.mtu)
+        if self.autonegotiate_admin:
+            port_obj.add_new_child(
+                "is-administrative-auto-negotiate", self.autonegotiate_admin)
+        if self.duplex_admin:
+            port_obj.add_new_child("administrative-duplex", self.duplex_admin)
+        if self.speed_admin:
+            port_obj.add_new_child("administrative-speed", self.speed_admin)
+        if self.flowcontrol_admin:
+            port_obj.add_new_child(
+                "administrative-flowcontrol", self.flowcontrol_admin)
+        self.server.invoke_successfully(port_obj, True)
+
+    def apply(self):
+        """
+        Run Module based on play book
+        """
+        changed = False
+        net_port_exists = False
+        results = netapp_utils.get_cserver(self.server)
+        cserver = netapp_utils.setup_na_ontap_zapi(
+            module=self.module, vserver=results)
+        netapp_utils.ems_log_event("na_ontap_net_port", cserver)
+        net_port_details = self.get_net_port()
+        if net_port_details:
```

Comment:
shouldn't it be an error if the port *doesn't* exist? We're talking hardware here, right?

Reply:
This is something we'll look at fixing in the new version.